### PR TITLE
Add redeploy endpoint

### PR DESCRIPTION
**Pull request by andreagrandi**

### CODE REVIEW

File: deploy/deploy.py [hunk 1]
```diff
@@ -102,3 +102,39 @@ def delete():
 
     # currently just text
     return jsonify({'text': response.stdout.decode('utf-8')})
+
+
+@app.route('/api/redeploy', methods=['POST'])
+@requires_auth
+def redploy():
```
Comment (davidread) [inline deploy/deploy.py]: "redploy" typo

File: deploy/deploy.py [hunk 2]
```diff
@@ -102,3 +102,39 @@ def delete():
...
+    try:
+        data['chart'] = 'rstudio'
+        response = commands.delete_chart(args=data)
+    except subprocess.CalledProcessError as e:
+        app.logger.error('Error calling deploy.sh: %s', str(e.output))
+        return Response('Error calling deploy', 500)
+
+    # Deploy the app again
+    data = dict(
+        fullname='redeploy',
```
Comment (davidread) [inline deploy/deploy.py]: Might be worth a comment to explain that fullname and email are not used for anything...

**Comment by davidread** [comment]
As discussed, you might put in a wait after the deletes, polling pod_statuses until the pod has gone, otherwise the subsequent deploy fails.

**Comment by andreagrandi** [comment]
Yeah, but "how"? Because one thing is to just put a delay, very easy to implement. A totally different thing is to poll pod_status. Not because calling pod_status is hard itself, but because you have to call it, check its result, call it again etc....a simple loop sounds quite bad as solution (imho). Any alternatives?

**Comment by andreagrandi** [comment]
p.s: the last solution I proposed yesterday, is completely different by the way. I proposed to completely get rid of "redeploy" endpoint and let the UI (or better the Django backend) call the "delete" endpoint first, then redirect on the deleting page etc.... with the difference that instead of redirecting to the "delete" page when it's finished, we redirect to the "deploy" page. In this way we reuse what we have without making a complicated thing in the sandbox-deploy backend

**Comment by davidread** [comment]
Sounds good to me 👍

**Pull request closed without merging**